specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-01-01-preview/AlertRuleRecommendations.json

```diff
@@ -0,0 +1,328 @@
+            "$ref": "./examples/AlertRuleRecommendations_GetByResource_MAC.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/alertRuleRecommendations": {
+      "get": {
+        "tags": [
+          "alertRuleRecommendations"
+        ],
+        "description": "Retrieve alert rule recommendations for a target type.",
+        "operationId": "AlertRuleRecommendations_ListByTargetType",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetTypeParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request for a list of alert rule recommendations",
+            "schema": {
+              "$ref": "#/definitions/AlertRuleRecommendationsListResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List alert rule recommendations for virtual machines at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_VM.json"
+          },
+          "List alert rule recommendations for Monitoring accounts at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_MAC.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ResourceUriParameter": {
+      "name": "resourceUri",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The identifier of the resource.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
+    },
+    "TargetTypeParameter": {
+      "name": "targetType",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The recommendations target type.",
+      "x-ms-parameter-location": "client"
+    }
+  },
+  "definitions": {
+    "AlertRuleRecommendationsListResponse": {
+      "type": "object",
+      "description": "List of alert rule recommendations.",
+      "required": [
+        "value"
+      ],
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AlertRuleRecommendationResource"
+          },
+          "description": "the values for the alert rule recommendations."
+        }
+      }
+    },
+    "AlertRuleRecommendationResource": {
```

Comment:
>AlertRuleRecommendationResource [](http://example.com/codeflow?start=5&length=31)

Derive it from ARM common-types TrackedResource or ProxyResource, instead of defining id, name, type etc. required properties explicitly.

That way, you won't have to make any changes in future when new properties are added or removed from ARM's resource schema.